### ISSUE TITLE
Add comments to all algorithms

### ISSUE THREAD

**Issue by raklaptudirm** [TheAlgorithms/Javascript#630]
One of the main purposes of this repository is education. But for that, a comprehensive explanation of the various algorithms along with the mentioned references is required. This is achieved through well-commented code, but that is missing from many algorithms. This issue tries to start work on commenting on all the algorithms and establishing a good format for commenting. We also should check all new Pull Requests for code quality and comments.

A format suggestion for commenting:
```js
/*
* Brief explanation of the given steps of the algorithm
* along with manual word wrapping.
*
* Using paragraphs, diagrams, and tables wherever
* required.
*
* ┌──────────┬───────┐
* |   Row 1  | Row 2 |
* ├──────────┼───────┤
* | Column 1 |  1,2  |
* ├──────────┼───────┤
* | Column 2 |  2,2  |
* └──────────┴───────┘
*
* References:
* List of references used in the above-given explanation
* from various sources.
* Each on a separate line.
* https://en.wikipedia.org/wiki/Box-drawing_character
*/
```

**Comment by cclauss** [comment]
Is this largely duplication of https://github.com/TheAlgorithms/Algorithms-Explanation ?

**Comment by raklaptudirm** [comment]
@cclauss That is still not an excuse to have uncommented code in my opinion. But I am open to suggestions.

**Comment by codefuncode** [comment]
> @cclauss That is still not an excuse to have uncommented code in my opinion. But I am open to suggestions. A discussion would be welcome.

True

**Comment by nimidsin** [comment]
/*
* Complete explanation of the given steps of the algorithm
* along with manual word wrapping.
*
* Using paragraphs, diagrams, and tables wherever
* required.
*
* ┌──────────┬───────┐
* |   Row 1  | Row 2 |
* ├──────────┼───────┤
* | Column 1 |  1,2  |
* ├──────────┼───────┤
* | Column 2 |  2,2  |
* └──────────┴───────┘
*
* References:
* List of references used in the above-given explanation
* from various sources.
* Each on a separate line.
* https://en.wikipedia.org/wiki/Box-drawing_character
*/

**Comment by cclauss** [comment]
In my opinion we are much better served with self-documenting function and variable names so that the entities and intents of each algorithm are as clear as possible without comments.  This means that comments should be reserved for areas where clear code is impossible.

Vertical lines and manual spacing seems like a waste of valuable developer time and focus.

**Comment by Sin-Sumit** [comment]
> In my opinion we are much better served with self-documenting function and variable names so that the entities and intents of each algorithm are as clear as possible without comments. This means that comments should be reserved for areas where clear code is impossible.
> 
> Vertical lines and manual spacing seems like a waste of valuable developer time and focus.

True, add comments where clear code is impossible. And for writing the algorithm use respective variables, array, data structure names, so it will be easy to understand and no need to write comments at every algorithm.

**Comment by appgurueu** [comment]
> In my opinion we are much better served with self-documenting function and variable names so that the entities and intents of each algorithm are as clear as possible without comments. This means that comments should be reserved for areas where clear code is impossible.
> 
> Vertical lines and manual spacing seems like a waste of valuable developer time and focus.

Well put. A full plain-english explanation is better left to the algorithms-explanation repo.
